cartographer_grpc/framework/server_test.cc

```diff
@@ -28,17 +28,65 @@ namespace framework {
 namespace {
 
 class GetServerOptionsHandler
-    : public RpcHandler<Stream<proto::Request>, proto::Response> {};
-
-TEST(ServerTest, StartServerTest) {
-  Server::Builder server_builder;
-  server_builder.SetServerAddress("0.0.0.0:50051");
-  server_builder.SetNumberOfThreads(1);
-  server_builder.RegisterHandler<GetServerOptionsHandler, proto::Math>(
-      "GetSum");
-  std::unique_ptr<Server> server = server_builder.Build();
-  server->Start();
-  server->Shutdown();
+    : public RpcHandler<Stream<proto::GetSumRequest>, proto::GetSumResponse> {
+ public:
+  void OnRequest(const proto::GetSumRequest& request) override {
+    sum_ += request.input();
+  }
+
+  void OnReadsDone() override {
+    auto response = cartographer::common::make_unique<proto::GetSumResponse>();
+    response->set_output(sum_);
+    Send(std::move(response));
+  }
+
+ private:
+  int sum_ = 0;
+};
+
+const std::string kServerAddress = "localhost:50051";
+const std::size_t kNumThreads = 1;
+
+class ServerTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    Server::Builder server_builder;
+    server_builder.SetServerAddress(kServerAddress);
```

Comment:
I think these tests become flaky if run in parallel because of the shared port.
If this is the case, add a todo.

Reply:
Good point. Done.